### ISSUE TITLE
Implement `is_closed()`?

### ISSUE THREAD

**Issue by mgeier** [mgeier/rtrb#41]
I've never needed this, but I've seen that it's implemented in:

https://github.com/Ralith/oddio/blob/edfec45801ea88f130cd0be719298b9c12673469/src/spsc.rs#L95-L99

and

https://github.com/Ralith/oddio/blob/edfec45801ea88f130cd0be719298b9c12673469/src/spsc.rs#L134-L138

I guess this could also be implemented with [Arc::strong_count()](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.strong_count).

Does anyone need this?

Are there any downsides to implementing this?

**Comment by mgeier** [comment]
I guess `is_closed()` could easily be implemented, but the question is whether writing to a `Producer` with a closed `Consumer` should return a specific error.
This would be a breaking change.

Reading from a `Consumer` with a closed `Producer` might still work if there are items left in the buffer, but once the buffer is empty, should the error say "empty" or "closed"?

I'm not sure if any of this makes sense, but if it does, it should probably be implemented in a separate set of `struct`s, e.g. `Channel`, `Sender` and `Receiver`.

**Comment by mgeier** [comment]
See also https://github.com/crossbeam-rs/crossbeam/pull/497, https://github.com/crossbeam-rs/crossbeam/issues/586.

**Comment by mgeier** [comment]
Trying to implement this, I've come to the conclusion that it doesn't really make sense to include this into the error types.

```rust
enum PushError<T> {
    Full(T),
    Closed(T),
}
```

The two variants would not be mutually exclusive, which makes it an awkward error type.
If the buffer is full, it might be closed at the same time and vice versa.

```rust
enum PopError {
    Empty,
    EmptyAndClosed,
}
```

This is even stranger, because if the buffer is `Closed` but not `Empty`, we are still able to `pop()` items and no error should happen.

Therefore, I've abandoned this idea and implemented only the `is_abandoned()` methods in #46.

Speaking of which, I think "abandoned" captures the state more clearly than "closed".
